### PR TITLE
Improve response log

### PR DESCRIPTION
**Pull request by ianlessa**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| What?         | Improve response log
| Why?          | To improve debug.
| How?          | Added curl error log and response raw log on null.

#### Status
READY

### CODE REVIEW

File: app/code/community/Uecommerce/Mundipagg/Model/Api.php [hunk 1]
```diff
@@ -1150,6 +1178,13 @@ public function getTransactionHistory($orderKey) {
         curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
         // Execute get
         $_response = curl_exec($ch);
+
+        //check for curl errors
+        $curlErrorNumber = curl_errno($ch);
```
Comment (felipebool) [inline app/code/community/Uecommerce/Mundipagg/Model/Api.php]: https://curl.haxx.se/libcurl/c/libcurl-errors.html
: )

File: app/code/community/Uecommerce/Mundipagg/Model/Api.php [hunk 2]
```diff
@@ -1278,10 +1313,23 @@ public function sendRequest($dataToPost, $url, $_logRequest = array()) {
...
+        }
+
         // Close connection
         curl_close($ch);
         // Is there an error?
         $xml = simplexml_load_string($_response);
+
+        // Check for invalid xml response
+        if($xml === false) {
+            $helperLog->error("INVALID XML RESPONSE!\n" . $_response . "\n\n");
```
Comment (felipebool) [inline app/code/community/Uecommerce/Mundipagg/Model/Api.php]: http://php.net/manual/en/function.libxml-get-errors.php
A gente podia mostrar a mensagem de erro no log

File: app/code/community/Uecommerce/Mundipagg/Model/Api.php [hunk 3]
```diff
@@ -874,7 +875,20 @@ public function manageOrderRequest($data, Uecommerce_Mundipagg_Model_Standard $s
...
+            //check for curl errors
+            $curlErrorNumber = curl_errno($ch);
+            if ($curlErrorNumber !== 0) {
+                $helperLog->error("CURL ERROR ". $curlErrorNumber . "!" . curl_error($ch));
+            }
+
             $xml = simplexml_load_string($_response);
+
+            // Check for invalid xml response
+            if($xml === false) {
```
Comment (michelpl) [inline app/code/community/Uecommerce/Mundipagg/Model/Api.php]: espaço

File: app/code/community/Uecommerce/Mundipagg/Model/Api.php [hunk 4]
```diff
@@ -925,10 +939,24 @@ public function saleCapture($data, $orderReference) {
...
+        if ($curlErrorNumber !== 0) {
+            $log->error("CURL ERROR ". $curlErrorNumber . "!" . curl_error($ch));
+        }
+
         // Close connection
         curl_close($ch);
+        $response = json_decode($response, true);
+
+        // Check for invalid json response
+        if($response === null) {
```
Comment (michelpl) [inline app/code/community/Uecommerce/Mundipagg/Model/Api.php]: espaço

File: app/code/community/Uecommerce/Mundipagg/Model/Api.php [hunk 5]
```diff
@@ -1278,10 +1313,23 @@ public function sendRequest($dataToPost, $url, $_logRequest = array()) {
...
+            $helperLog->error("CURL ERROR ". $curlErrorNumber . "!" . curl_error($ch));
+        }
+
         // Close connection
         curl_close($ch);
         // Is there an error?
         $xml = simplexml_load_string($_response);
+
+        // Check for invalid xml response
+        if($xml === false) {
```
Comment (michelpl) [inline app/code/community/Uecommerce/Mundipagg/Model/Api.php]: espaço

File: app/code/community/Uecommerce/Mundipagg/Model/Api.php [hunk 6]
```diff
@@ -1342,10 +1390,23 @@ public function sendJSON($data) {
...
+            $log->error("CURL ERROR ". $curlErrorNumber . "!" . curl_error($ch));
+        }
+
         // Close connection
         curl_close($ch);
+
         $responseData = json_decode($response, true);
 
+        // Check for invalid json response
+        if($responseData === null) {
```
Comment (michelpl) [inline app/code/community/Uecommerce/Mundipagg/Model/Api.php]: espaço

**Pull request merged**